app/models/peep.rb

```diff
@@ -0,0 +1,19 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+
+class Peep
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :message, String
+  property :post_time, Text
+  property :user_id, Serial
+  property :username, String
```

Comment:
Don't store `username` in peep!  The username is in the users table and is made accessible to the peep through the association.